component/portal/src/main/java/io/meeds/portal/security/service/SecuritySettingService.java

```diff
@@ -0,0 +1,174 @@
+/**
+ * This file is part of the Meeds project (https://meeds.io/).
+ *
+ * Copyright (C) 2020 - 2023 Meeds Association contact@meeds.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package io.meeds.portal.security.service;
+
+import static io.meeds.portal.security.constant.UserRegistrationType.OPEN;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+
+import org.exoplatform.commons.api.settings.SettingService;
+import org.exoplatform.commons.api.settings.SettingValue;
+import org.exoplatform.commons.api.settings.data.Context;
+import org.exoplatform.commons.api.settings.data.Scope;
+
+import io.meeds.portal.security.constant.UserRegistrationType;
+import io.meeds.portal.security.model.RegistrationSetting;
+
+public class SecuritySettingService {
+
+  protected static final String               INTERNAL_USERS_GROUP               = "/platform/users";
+
+  protected static final String               EXTERNAL_USERS_GROUP               = "/platform/externals";
+
+  protected static final Context              SECURITY_CONTEXT                   = Context.GLOBAL.id("SECURITY");
+
+  protected static final Scope                SECURITY_SCOPE                     = Scope.APPLICATION.id("SECURITY");
+
+  protected static final String               REGISTRATION_TYPE_PARAM            = "REGISTRATION_TYPE";
+
+  protected static final String               REGISTRATION_EXTERNAL_USER_PARAM   = "REGISTRATION_EXTERNAL_USER";
+
+  protected static final String               REGISTRATION_EXTRA_GROUPS_PARAM    = "REGISTRATION_EXTRA_GROUPS";
+
+  protected static final String               EXTRA_GROUPS_SEPARATOR             = ",";
+
+  protected static final UserRegistrationType DEFAULT_REGISTRATION_TYPE          =
+                                                                        UserRegistrationType.valueOf(System.getProperty("meeds.security.registration.type",
```

Comment:
I use the property meeds.security.registration.type to set by default to restricted.
meeds.security.registration.type=restricted

It generate an error at startup : 

> Caused by: java.lang.IllegalArgumentException: No enum constant io.meeds.portal.security.constant.UserRegistrationType.restricted
> 	at java.base/java.lang.Enum.valueOf(Enum.java:273)
> 	at io.meeds.portal.security.constant.UserRegistrationType.valueOf(UserRegistrationType.java:22)
> 	at io.meeds.portal.security.service.SecuritySettingService.<clinit>(SecuritySettingService.java:56)
> 	... 82 common frames omitted
> 

I had to change to 
meeds.security.registration.type=RESTRICTED

As all our properties are not capitalized, could you modify to accept 'restricted' ?

Reply:
Ah yes, it must be case insensitive. Done.
BTW, the property names has been changed to fit the ones specified in MIP. Thanks for your help.